codegen/smithy-aws-typescript-codegen/src/main/java/software/amazon/smithy/aws/typescript/codegen/RestJsonProtocolGenerator.java

```diff
@@ -90,6 +90,12 @@ protected void writeDefaultHeaders(GenerationContext context, OperationShape ope
         AwsProtocolUtils.generateUnsignedPayloadSigV4Header(context, operation);
     }
 
+    @Override
+    protected void writeDefaultErrorHeaders(GenerationContext context, StructureShape error) {
+        super.writeDefaultErrorHeaders(context, error);
+        context.getWriter().write("'x-amzn-errortype': $S,", error.getId().getName());
+    }
```

Comment:
Is this used anywhere?

Reply:
Yes, this is used for the ssdk when serializing errors